### PR TITLE
Clarify the terms "bag" and "options" for the DateTimeFormat docs

### PR DESCRIPTION
**Pull request by gregtatum**
I found myself confused by these terms, and how they worked. I updated the docs to try and clarify what these represent, and to clarify the state of the future planned work. It took me a bit of time to figure out that Components wasn't actually implemented yet, and so this is an attempt to add more clarity.

Please make sure my assumptions here on what I read are correct, and that the future planned work is accurately represented. I didn't know if it was worth making sure the future planned work had links to issues on them.

### CODE REVIEW

File: components/datetime/src/options/components.rs
Comment (gregtatum) [inline components/datetime/src/options/components.rs:20]: I still don't really understand what `skeleton` means here still. It's the first use of this term here without a definition.

Comment (zbraniecki) [reply components/datetime/src/options/components.rs:20]: Great catch! It's a bit documented here: https://unicode.org/reports/tr35/tr35-dates.html#availableFormats_appendItems but I feel like Unicode generally underdocuments skeletons.

If I were to describe it, I'd go for something like:

```
Skeletons are locale-agnostic identifiers containing field information, and in a canonical order. Examples are "yMMMM" for year + full month, or "MMMd" for abbreviated month + day.
```

Comment (gregtatum) [reply components/datetime/src/options/components.rs:20]: > canonical order

It seems like skeleton is the incorrect term here then, as the components doesn't provide an order for these terms, it's an unordered collection.

Comment (zbraniecki) [reply components/datetime/src/options/components.rs:20]: The canonicality (is it a word?) of the order is only necessary because skeleton is a string. When representing a bag like:

```
{
  "year": "long",
  "month": "2-digit",
  "day": "2-digit"
}
```

as a bag, I can set any order, and the output will be the same. But when encoding it as a skeleton, we need to make sure that we don't end up with `yyyyMMDD` being different from `MMyyyyDD` and `DDMMyyyy`.
So when encoding skeletons as strings we use "canonical" order of year->month->day, and in result every skeleton will be a string encoded with fields in that order.

So, bag is a representation of a skeleton. It is unordered, but its hashing/stringification will be ordered.

Does that make sense?

**Review by zbraniecki — APPROVED** [review]
lgtm!

File: components/datetime/src/options/components.rs
Comment (sffc) [reply components/datetime/src/options/components.rs:20]: I use the term "skeleton" mostly interchangeably with what Zibi is defining as "components".  A skeleton can be represented either as a string (UTS 35 style) or as an options bag.  With this mental model, we could consider the Components class as having a FromStr that parses the UTS 35-style skeleton string.

File: components/datetime/src/options/mod.rs
Comment (sffc) [inline components/datetime/src/options/mod.rs:32]: I don't think we should separate Components and Skeleton since they are different representations of the same thing.

**Comment by gregtatum** [comment]
I'm holding off on applying the final feedback as I'm ramping up my knowledge on #272.

File: components/datetime/src/options/components.rs
Comment (mihnita) [reply components/datetime/src/options/components.rs:20]: > Skeletons are locale-agnostic identifiers containing field information in a canonical order.

The order does not matter.
"yMMMd" and "MMMyd" is the same skeleton, with the same result.

You can also "sprinkle" other characters, and they will be ignored:
"MMMM d, y" is a valid skeleton (and is the same as "yMMMMd")
But I don't like it.
I prefer ISO order and no skeleton, not a pattern.

**Comment by gregtatum** [comment]
I think I'm going to close this, and cherry-pick the useful bits after landing the components bag work. I'll open a new PR when I do that.

**Pull request closed without merging**
